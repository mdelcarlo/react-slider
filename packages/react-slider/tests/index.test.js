import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import serializer from 'jest-emotion';
import Slider from '../index';

expect.addSnapshotSerializer(serializer);

test('snapshot - defaults', () => {
  const { container } = render(
    <Slider
      {...{
        min: -100,
        max: 100,
        step: 5,
      }}
    />
  );
  expect(container).toMatchSnapshot();
});

test('snapshot - disabled', () => {
  const { container } = render(
    <Slider
      {...{
        min: 0,
        max: 100,
        disabled: true,
      }}
    />
  );
  expect(container).toMatchSnapshot();
});

test('snapshot - labeled', () => {
  const { container } = render(
    <Slider
      {...{
        min: 0,
        max: 100,
        step: 10,
        value: 100,
        showLabel: true,
      }}
    />
  );
  expect(container).toMatchSnapshot();
});

test('snapshot - hasVisibleSteps', () => {
  const { container } = render(
    <Slider
      {...{
        min: 0,
        max: 100,
        step: 10,
        value: 100,
        hasVisibleSteps: true,
      }}
    />
  );
  expect(container).toMatchSnapshot();
});
