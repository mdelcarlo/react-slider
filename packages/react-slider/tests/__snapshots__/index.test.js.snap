// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot - defaults 1`] = `
.emotion-0 {
  left: calc(0% - 8px);
}

.emotion-0 {
  left: calc(0% - 8px);
}

<div>
  <div
    aria-labelledby="slider__label"
    class="slider emotion-1"
    role="slider"
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="-100"
      aria-valuenow="-100"
      aria-valuetext="-100"
      class="slider__thumb emotion-0"
      focusstyle="[object Object]"
      max="100"
      min="-100"
      position="0"
      role="slider"
      selectedvalue="-100"
      tabindex="1"
    />
  </div>
</div>
`;

exports[`snapshot - disabled 1`] = `
.emotion-0 {
  left: calc(0% - 8px);
}

.emotion-0 {
  left: calc(0% - 8px);
}

<div>
  <div
    aria-disabled="true"
    aria-labelledby="slider__label"
    aria-readonly="true"
    class="slider emotion-1"
    disabled=""
    role="slider"
  >
    <div
      aria-disabled="true"
      aria-orientation="horizontal"
      aria-readonly="true"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      aria-valuetext="0"
      class="slider__thumb emotion-0"
      disabled=""
      focusstyle="[object Object]"
      max="100"
      min="0"
      position="0"
      role="slider"
      selectedvalue="0"
      tabindex="-1"
    />
  </div>
</div>
`;

exports[`snapshot - hasVisibleSteps 1`] = `
.emotion-0 {
  left: calc(100% - 8px);
}

.emotion-1 {
  left: calc(0%);
}

.emotion-2 {
  left: calc(10%);
}

.emotion-3 {
  left: calc(20%);
}

.emotion-4 {
  left: calc(30%);
}

.emotion-5 {
  left: calc(40%);
}

.emotion-6 {
  left: calc(50%);
}

.emotion-7 {
  left: calc(60%);
}

.emotion-8 {
  left: calc(70%);
}

.emotion-9 {
  left: calc(80%);
}

.emotion-10 {
  left: calc(90%);
}

.emotion-11 {
  left: calc(100%);
}

.emotion-0 {
  left: calc(100% - 8px);
}

.emotion-1 {
  left: calc(0%);
}

.emotion-2 {
  left: calc(10%);
}

.emotion-3 {
  left: calc(20%);
}

.emotion-4 {
  left: calc(30%);
}

.emotion-5 {
  left: calc(40%);
}

.emotion-6 {
  left: calc(50%);
}

.emotion-7 {
  left: calc(60%);
}

.emotion-8 {
  left: calc(70%);
}

.emotion-9 {
  left: calc(80%);
}

.emotion-10 {
  left: calc(90%);
}

.emotion-11 {
  left: calc(100%);
}

<div>
  <div
    aria-labelledby="slider__label"
    class="slider emotion-12"
    role="slider"
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      aria-valuetext="100"
      class="slider__thumb emotion-0"
      focusstyle="[object Object]"
      max="100"
      min="0"
      position="100"
      role="slider"
      selectedvalue="100"
      tabindex="1"
    />
    <a
      class="slider__step emotion-1"
    />
    <a
      class="slider__step emotion-2"
    />
    <a
      class="slider__step emotion-3"
    />
    <a
      class="slider__step emotion-4"
    />
    <a
      class="slider__step emotion-5"
    />
    <a
      class="slider__step emotion-6"
    />
    <a
      class="slider__step emotion-7"
    />
    <a
      class="slider__step emotion-8"
    />
    <a
      class="slider__step emotion-9"
    />
    <a
      class="slider__step emotion-10"
    />
    <a
      class="slider__step emotion-11"
    />
  </div>
</div>
`;

exports[`snapshot - labeled 1`] = `
.emotion-0 {
  left: calc(100% - 8px);
}

.emotion-0 {
  left: calc(100% - 8px);
}

<div>
  <div
    aria-labelledby="slider__label"
    class="slider emotion-1"
    role="slider"
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      aria-valuetext="100"
      class="slider__thumb emotion-0"
      focusstyle="[object Object]"
      max="100"
      min="0"
      position="100"
      role="slider"
      selectedvalue="100"
      tabindex="1"
    >
      <div
        class="slider__label"
        id="slider__label"
      >
        100
      </div>
    </div>
  </div>
</div>
`;
